docs/elasticsearch-spec.asciidoc

```diff
@@ -101,6 +101,73 @@ $ openssl req -x509 -newkey rsa:4096 -keyout tls.key -out tls.crt -days 365 -nod
 $ kubectl create secret tls my-cert --cert tls.crt --key tls.key
 ----
 
+[id="{p}-es-secure-settings"]
+=== Secure Settings
+
+link:https://www.elastic.co/guide/en/elasticsearch/reference/current/secure-settings.html[Secure settings] can be specified via a Kubernetes secret.
+The secret should contain a key-value pair for each secure setting you want to add. Reference that secret in the Elasticsearch
+resource spec for ECK to automatically inject those settings into the keystore on each node before it starts Elasticsearch.
+
+[source,yaml]
+----
+spec:
+  secureSettings:
+    secretName: your-secure-settings-secret
+----
+
+See link:snapshots.asciidoc[How to create automated snapshots] for an example use case.
+
+
+[id="{p}-bundles-plugins"]
+=== Custom Configuration Files and Plugins
+
+To run Elasticsearch with specific plugins or configurations files installed on ECK you have two options:
+
+1. create a custom Docker image with the plugins or files pre-installed
+2. install the plugins or configuration files at pod startup time
+
+NOTE: The first option has the advantage that you can verify the correctness of the image before rolling it out to your ECK installation, while the second option gives you
+maximum flexibility. But the second options also means you might catch any errors only at runtime. Plugin installation at runtime has another drawback in that it needs access to the Internet from your cluster
+and downloads each plugin multiple times, once for each Elasticsearch node.
+
+The following describes option 2 using a repository plugin as the example. To install the plugin before the Elasticsearch
```

Comment:
Presented as such, option 1 seems to be a better choice and we choose to describe option 2. Should we at least also describe option 1 and highlight that option 1 requires to have a registry and to manage the lifecycle of the custom image and ?